server/internal/k8s/k8sclient.go

```diff
@@ -11,12 +11,16 @@ import (
 )
 
 // Client is a wrapper for Kubernetes clientsets.
-type Client struct {
+type Client interface {
```

Comment:
My preference is to define the interface when it is used as much as possible. This is not always better as it might cause redundant definitions, but https://go.dev/wiki/CodeReviewComments#interfaces has the following recommendation:

> Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values. The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.

For this case, I think we can define `podlister` interface where `Client` is consumed (= `refreshRoutes`).